tests/cbmc/proofs/s2n_stuffer_skip_whitespace/s2n_stuffer_skip_whitespace_harness.c

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "api/s2n.h"
+#include "stuffer/s2n_stuffer.h"
+
+#include <assert.h>
+#include <cbmc_proof/cbmc_utils.h>
+#include <cbmc_proof/make_common_datastructures.h>
+#include <cbmc_proof/proof_allocators.h>
+
+void s2n_stuffer_skip_whitespace_harness() {
+    /* Non-deterministic inputs. */
+    struct s2n_stuffer *stuffer = cbmc_allocate_s2n_stuffer();
+    __CPROVER_assume(s2n_stuffer_is_valid(stuffer));
+    __CPROVER_assume(s2n_blob_is_bounded(&stuffer->blob, MAX_BLOB_SIZE));
+    uint32_t skipped;
+
+    /* Save previous state from stuffer. */
+    struct s2n_stuffer old_stuffer = *stuffer;
+    struct store_byte_from_buffer old_byte_from_stuffer;
+    save_byte_from_blob(&stuffer->blob, &old_byte_from_stuffer);
+
+    /* Operation under verification. */
+    if(s2n_stuffer_skip_whitespace(stuffer, &skipped) == S2N_SUCCESS) {
+        if (skipped > 0) {
+            assert(stuffer->read_cursor == old_stuffer.read_cursor + skipped);
+            size_t index;
+            __CPROVER_assume(index >= old_stuffer.read_cursor && index < stuffer->read_cursor);
+            assert(stuffer->blob.data[index] == ' '  ||
+                   stuffer->blob.data[index] == '\t' ||
+                   stuffer->blob.data[index] == '\n' ||
+                   stuffer->blob.data[index] == '\r');
+        }
```

Comment:
If nothing was skipped, should we verify that an index is not a whitespace character?

Reply:
Done.